napari/_qt/utils.py

```diff
@@ -469,3 +471,14 @@ def qt_might_be_rich_text(text) -> bool:
         return _Qt.mightBeRichText(text)
     except Exception:
         return bool(RICH_TEXT_PATTERN.search(text))
+
+
+def check_if_in_main_thread():
```

Comment:
Nit: when I see `check` I think of functions that will `raise` if a condition isn't met. Instead this returns a boolean, in which case something shorter is probably more clear and less verbose.

```suggestion
def in_main_thread() -> bool:
```